lib/include/defines.h

```diff
@@ -55,8 +55,15 @@ inline const char* get_kernelName(const int integrator,
     case SIXTH:
       if(precision == DOUBLESINGLE)
       {
+#ifdef _OCL_
+          fprintf(stderr, "ERROR: Sixth order integrator with double single precision");
+          fprintf(stderr, "ERROR: is not implemented in OpenCL, only in CUDA. Please");
+          fprintf(stderr, "ERROR: file an issue on GitHub if you need this combination.");
+          exit(1);
+#else
```

Comment:
I guess there's no need for the else? Given that there's the exit above? In that case maybe change the `#else` into `#endif`

Reply:
The problem here actually was that `float3` was undefined when compiling with OpenCL, causing a compiler error. So we could move the `perThreadSM = ...` line outside of the `#ifdef`, but then it wouldn't compile for OpenCL.

I think I later saw a header somewhere that aliases some OpenCL types to CUDA-like names, so maybe `float3` can be added there to fix it instead, I'll have a look.